### PR TITLE
additional DSN options for SQLServer 

### PR DESCRIPTION
**Pull request by stmeyer**
fix for #9963. This makes the use of SQL Server more flexible.

### CODE REVIEW

File: tests/TestCase/Database/Driver/SqlserverTest.php [hunk 1]
```diff
@@ -67,6 +67,12 @@ public function testConnectionConfigCustom()
             PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION,
             PDO::SQLSRV_ATTR_ENCODING => 'a-language'
         ];
+        $expected['attributes'] = [];
+        $expected['app'] = null;
+        $expected['connectionPooling'] = null;
+        $expected['failoverPartner'] = null;
+        $expected['loginTimeout'] = null;
+        $expected['multiSubnetFailover'] = null;
```
Comment (markstory) [inline tests/TestCase/Database/Driver/SqlserverTest.php]: Could we have a test where these options are added to the DSN string?

**Comment by stmeyer** [comment]
@markstory do you have a hint how to get the `$dns` for testing? It is set inside the method https://github.com/cakephp/cakephp/pull/10490/files#diff-6d58c9b7093fe3eba95388ad543a0da1R61

**Comment by markstory** [comment]
@stmeyer You could use a mocked driver instance that mocks `_connect`

```php
        $driver = $this->getMockBuilder('Cake\Database\Driver\Sqlserver')
            ->setMethods(['_connect'])
            ->setConstructorArgs([[]])
```

This would let you set expectations, and call `connect()` which then calls `_connect` with the DSN string.
            ->getMock();

**Comment by stmeyer** [comment]
Travis-CI stopped working:
```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received

The build has been terminated
```

Can you please restart the tests?

File: tests/TestCase/Database/Driver/SqlserverTest.php [hunk 2]
```diff
@@ -75,15 +137,13 @@ public function testConnectionConfigCustom()
         $expected['multiSubnetFailover'] = null;
 
         $connection = $this->getMockBuilder('stdClass')
-            ->setMethods(['exec', 'quote'])
-            ->getMock();
+                ->setMethods(['exec', 'quote'])
+                ->getMock();
```
Comment (markstory) [inline tests/TestCase/Database/Driver/SqlserverTest.php]: Why the extra indentation here?

**Review by markstory — COMMENTED** [review]
Outside of code style errors, this looks good.

File: tests/TestCase/Database/Driver/SqlserverTest.php [hunk 3]
```diff
@@ -36,6 +37,67 @@ public function setUp()
...
+        ];
+    }
+
+    /**
+     * Test if all options in dns string are set
+     *
+     * @dataProvider dnsStringDataProvider
+     * @param array $constructorArgs
+     * @param string $dnsString
+     *
```
Comment (ravage84) [inline tests/TestCase/Database/Driver/SqlserverTest.php]: No white line here.

**Review by ravage84 — COMMENTED** [review]
Nitpicking from my side.

**Pull request merged**